src/ShallowRenderer.ts

```diff
@@ -43,23 +54,12 @@ export default class ShallowRenderer implements AbstractShallowRenderer {
         return;
       }
 
-      // Monkey-patch the component's `render` to make it shallow-render.
-      const instance = rootNode.instance;
-      const originalRender = instance.render;
-      instance.render = function (...args: any[]) {
-        let result;
-        withShallowRendering(() => {
-          result = originalRender.call(this, ...args);
-        });
-        return result;
-      };
```

Comment:
I moved this functionality into the `patchShallowRoot` method so we have only one place where we override the render method.